### ISSUE TITLE
Nested Fieldsets from cdf?

### ISSUE THREAD

**Issue by conkline** [OceanParcels/parcels#673]
I have two sets of netcdf grids - they have the same timestep and same number of files, however they are two different resolutions and two different spatial scales (a smaller, finer scale grid inside a large, coarser scale grid). I can load each of these as fieldsets with the command:

`fieldsetFine = FieldSet.from_netcdf("/path_to_files/*cdf", variables=var, dimensions=dim, deferred_load=True, timestamps=ts)`

However, I am at a loss on how to build a fieldset of nested fields, from netcdf files. I have ~800 odd files so I would need to keep the deferred_load behavior. Iʻd also love to use NestedField objects somehow, but Iʻm not sure this is possible. Any thoughts would be much appreciated!

**Comment by erikvansebille** [comment]
Have you had a look at https://nbviewer.jupyter.org/github/OceanParcels/parcels/blob/master/parcels/examples/tutorial_NestedFields.ipynb? 

Note that you can also use `Field.from_netcdf` in cells 2 and 3. 

See also https://github.com/OceanParcels/Parcelsv2.0PaperNorthSeaScripts/blob/master/run_northsea_mp.py#L82-L83

**Comment by conkline** [comment]
Hi Erik!

Yes, I have seen the tutorial, thanks. I am still having some issues getting NestedFields to work. On trying to read in a field from cdf, I get the following:

```
dim = {'lon':'Longitude_u', 'lat':'Latitude_v', 'depth':'Depth_w'}
ts = np.arange(0, 73612800, 86400)
fieldCoarse = Field.from_netcdf(glob.glob("/10tb_leviathan/tmp_4km/*cdf"), variable="U", dimensions=dim, timestamps=ts)

ValueError: zero-dimensional arrays cannot be concatenated
```

Not sure what I am doing wrong here - instead of digging into it too deeply, I have tried to simply pull the fields from my existing fieldsets, and add them to a current fieldset a la the North Sea code, but that doesnʻt work either. 
```
  UFine = fieldsetFine.get_fields()[0]
  VFine = fieldsetFine.get_fields()[1]
  WFine = fieldsetFine.get_fields()[2]
  UCoarse = fieldsetCoarse.get_fields()[0]
  VCoarse = fieldsetCoarse.get_fields()[1]
  WCoarse = fieldsetCoarse.get_fields()[2]
  
  UNested = NestedField('U', [UFine, UCoarse])
  VNested = NestedField('V', [VFine, VCoarse])
  WNested = NestedField('W', [WFine, WCoarse])
  
  fieldsetNested = fieldsetCoarse
  fieldsetNested.U = UNested
  fieldsetNested.V = VNested
  fieldsetNested.W = WNested
```

When I try to execute using this fieldset, it compiles, but I get the following error:
```
INFO: Compiled LarvalParticleAdvectionRK4_3DRWDiffusion ==> /tmp/parcels-1000/efe43a27800134537c73374ab846409b.so

---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-274-79cd2e79add1> in <module>
----> 1 pset.execute(AdvectionRK4_3D + k_RWDiffusion, runtime=duration, dt=timedelta(minutes=30))

~/miniconda3/envs/py3_parcels/lib/python3.7/site-packages/parcels/particleset.py in execute(self, pyfunc, endtime, runtime,dt, moviedt, recovery, output_file, movie_background_field, verbose_progress)
    399         next_output = time + outputdt if dt > 0 else time - outputdt
    400         next_movie = time + moviedt if dt > 0 else time - moviedt
--> 401         next_input = self.fieldset.computeTimeChunk(time, np.sign(dt))
    402
    403         tol = 1e-12

~/miniconda3/envs/py3_parcels/lib/python3.7/site-packages/parcels/fieldset.py in computeTimeChunk(self, time, dt)
    747             if type(f) in [VectorField, NestedField, SummedField] or not f.grid.defer_load:
    748                 continue
--> 749             if f.grid.update_status == 'not_updated':
    750                 nextTime_loc = f.grid.computeTimeChunk(f, time, signdt)
    751             nextTime = min(nextTime, nextTime_loc) if signdt >= 0 else max(nextTime, nextTime_loc)

AttributeError: 'RectilinearZGrid' object has no attribute 'update_status'
```

Obviously I am doing something wrong, but am not sure yet what it is. Any help is much appreciated.

Thanks,
Emily

**Comment by erikvansebille** [comment]
Thanks Emily. Can you give me a bit more info on the first error (the `ValueError`)? What is the full error message? And would it be possible for you to send me a part of the cdf files via e.g. WeShare? My email address  is e.vansebille@uu.nl

The second approach is a bit more hacky, so ideally we can fix the first one and you don't have to worry. We haven't tested nested fields too extensively yet, so it may indeed be that not all use cases work, just bear with us!

**Comment by conkline** [comment]
Hi Erik, so sorry for not getting back to you in a timely fashion! Here's a slightly more complete code chunk:

```
dim = {'lon':'Longitude_u', 'lat':'Latitude_t', 'depth':'Depth_w'}
ts = np.arange(0, 73612800, 86400)

fieldCoarse = Field.from_netcdf(glob.glob("/10tb_leviathan/tmp_4km/*cdf"), variable="U", dimensions=dim, timestamps=ts)
```

And the complete error message:
```
Traceback (most recent call last):

  File "<ipython-input-25-71fa428735d7>", line 1, in <module>
    fieldCoarse = Field.from_netcdf(glob.glob("/10tb_leviathan/tmp_4km/*cdf"), variable="U", dimensions=dim, timestamps=ts)

  File "/home/emily/miniconda3/envs/py3_parcels/lib/python3.7/site-packages/parcels/field.py", line 268, in from_netcdf
    time = np.concatenate(timeslices)

ValueError: zero-dimensional arrays cannot be concatenated
```
I'm guessing it has something to do with the timestamps, but I'm not sure. I'll also send along example cdf files shortly, thanks again for your help!!

**Comment by erikvansebille** [comment]
HI @conkline, thanks for the update and for sharing the data file. I've now been able to look at your Issue

First of all, you seem to be running a fairly old version of Parcels? In the latest Parcels, the `variable` has to be a tuple, like `('u': 'u')`. 

Also, the timestamps needs to be a nested list now: the outer list is the number of netcdf files, and the inner lists are the timestamps in each netcdf file. SO in the example that you send, it would be `ts = [[86400], [1.190592e+08]]`

Then, there was a bug with files that have only one snapshot in them, that has now been fixed in PR #705 

A code that works on my system (with PR #705) is
```python
from parcels import Field
import glob

dim = {'lon': 'Longitude_u', 'lat': 'Latitude_t', 'depth': 'Depth_w'}
ts = [[86400], [1.190592e+08]]

fieldCoarse = Field.from_netcdf(glob.glob("/Users/erik/Downloads/wetransfer-359e46/*cdf"), variable=('U', 'u'),
                                dimensions=dim, timestamps=ts)
```
